sdk/identity/identity/review/identity.api.md

```diff
@@ -152,11 +152,20 @@ export class DefaultAzureCredential extends ChainedTokenCredential {
 }
 
 // @public
-export interface DefaultAzureCredentialOptions extends TokenCredentialOptions {
+export interface DefaultAzureCredentialClientIdOptions extends TokenCredentialOptions {
     managedIdentityClientId?: string;
     tenantId?: string;
 }
 
+// @public
+export type DefaultAzureCredentialOptions = DefaultAzureCredentialClientIdOptions | DefaultAzureCredentialResourceIdOptions;
+
+// @public
+export interface DefaultAzureCredentialResourceIdOptions extends TokenCredentialOptions {
```

Comment:
To my eyes, this type name implies that `tenantId` is related to the resource ID

Reply:
Tell me more 🤔

Reply:
The name `DefaultAzureCredentialResourceIdOptions` suggests, to me at least, that all this type's fields pertain to a resource ID, not merely that it carries a resource ID in contrast to the type which carries a client ID.

Reply:
ohh I see. So, that is specifically because of the “ResourceIdOptions” part. How about `DefaultAzureCredentialOptionsWithResourceId`?

Reply:
Pushed here: https://github.com/Azure/azure-sdk-for-js/pull/20043/commits/27e8b920b6de4d5784db4448b27e80e005c3ccf0